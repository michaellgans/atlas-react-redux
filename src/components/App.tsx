function App() {
  return <>Hello Atlas</>;
}

export default App;
