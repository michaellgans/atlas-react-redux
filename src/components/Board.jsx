// Board Component

// Asset Imports

// Script Imports

// Returns a Board Component
export function Board() {
    // Define Hook
    return (
    );
}
