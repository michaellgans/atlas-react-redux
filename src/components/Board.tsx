// Board Component

// Asset Imports
import { useAppSelector } from "../store";
import { List } from "./List";
import { RootState } from "../store";

// Returns a Board Component
export function Board() {
  // Define Hook
  const lists = useAppSelector((state: RootState) => state.listsSlice.items);

  return (
    // START BOARD
    <div className="m-auto h-screen w-screen overflow-x-scroll text-center">
      <div className="flex h-full space-x-4">
        {lists.map((list) => (
          <List key={list.id} listID={list.id} listTitle={list.title} />
        ))}
      </div>
    </div>
  );
}
