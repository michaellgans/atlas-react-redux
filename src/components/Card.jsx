// Card Component

// Asset Imports

// Script Imports

// Returns a Card Component
export function Card() {
    // Define Hook
    return (
    );
}
