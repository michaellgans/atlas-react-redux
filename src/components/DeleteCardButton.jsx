// DeleteCardButton Component

// Asset Imports

// Script Imports

// Returns a DeleteCardButton Component
export function DeleteCardButton() {
    // Define Hook
    return (
    );
}
