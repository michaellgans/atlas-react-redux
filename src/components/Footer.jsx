// Footer Component

// Asset Imports

// Script Imports

// Returns a Footer Component
export function Footer() {
    // Define Hook
    return (
    );
}
