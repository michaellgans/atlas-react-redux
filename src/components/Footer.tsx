// Footer Component

// Imports
import { addList, clearBoard } from "../slices/listsSlice";
import { useAppDispatch } from "../store";
import { useState, FormEvent } from "react";

// Returns a Footer Component
export function Footer() {
  // Define Hook
  const [newList, setNewList] = useState("");
  const dispatch = useAppDispatch();

  const handleAddList = (event: FormEvent) => {
    // Prevent "submit" from reloading page
    event.preventDefault();
    if (newList.trim() !== "") {
      dispatch(addList(newList));
      setNewList("");
    }
  };

  const handleClearBoard = () => {
    console.log("Clear board was pressed");
    dispatch(clearBoard());
  };

  return (
    <footer className="sticky bottom-0 left-0 flex w-screen items-center justify-center space-x-8 border-t-2 border-blue bg-off-white-light p-8">
      <form onSubmit={handleAddList}>
        <input
          type="text"
          value={newList}
          placeholder="List title"
          onChange={(event) => setNewList(event.target.value)}
          name="title"
          className="border-0 bg-transparent text-3xl font-semibold text-blue placeholder:text-blue placeholder:opacity-50 focus:outline-none"
        />
        <button
          type="submit"
          className="mr-3 rounded bg-teal px-6 py-4 text-xl font-semibold text-off-white-light"
        >
          Save
        </button>
        <button
          type="button"
          onClick={handleClearBoard}
          className="rounded bg-teal px-6 py-4 text-xl font-semibold text-off-white-light"
        >
          Clear Board
        </button>
      </form>
    </footer>
  );
}
