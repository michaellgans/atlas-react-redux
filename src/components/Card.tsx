// Card Component

// Asset Imports
import { DeleteCardButton } from "./DeleteCardButton";

// Script Imports

// Types
type CardProps = {
  title: string;
  description: string;
};

// Returns a Card Component
export function Card({ title, description }: CardProps) {
  // Define Hook
  return (
    <div className="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue">
      <h5 className="my-2 flex w-full items-end justify-between text-xl font-black">
        <span>{title}</span>
        {/* DELETE CARD BUTTON */}
        <DeleteCardButton />
      </h5>
      <p className="mt-0 text-left">{description}</p>
    </div>
  );
}
