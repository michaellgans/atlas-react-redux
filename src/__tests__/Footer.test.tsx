// Test for Footer Component

// Imports
import { render } from "@testing-library/react";
import { expect, test, it } from 'vitest';
import { Footer } from "../components/Footer";

it('Controll Test', () => {
    expect(2 + 2).toBe(4);
});

test('Footer renders correctly', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
});
