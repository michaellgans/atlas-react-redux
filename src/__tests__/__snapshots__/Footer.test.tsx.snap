// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer renders correctly 1`] = `
<div>
  <footer
    class="sticky bottom-0 left-0 flex w-screen items-center justify-center space-x-8 border-t-2 border-blue bg-off-white-light p-8"
  >
    <form>
      <input
        class="border-0 bg-transparent text-3xl font-semibold text-blue placeholder:text-blue placeholder:opacity-50 focus:outline-none"
        name="title"
        placeholder="List title"
        type="text"
      />
      <button
        class="mr-3 rounded bg-teal px-6 py-4 text-xl font-semibold text-off-white-light"
        type="submit"
      >
        Save
      </button>
      <button
        class="rounded bg-teal px-6 py-4 text-xl font-semibold text-off-white-light"
        type="button"
      >
        Clear Board
      </button>
    </form>
  </footer>
</div>
`;
